api/server/server.go

```diff
@@ -137,6 +139,13 @@ func (s *Server) makeHTTPHandler(handler httputils.APIFunc) http.HandlerFunc {
 		if err := handlerFunc(ctx, w, r, vars); err != nil {
 			logrus.Errorf("Handler for %s %s returned error: %v", r.Method, r.URL.Path, err)
 			httputils.WriteError(w, err)
+			// Work around the client getting a "broken pipe" error
+			// and not reading an error response, which it needs to
+			// be able to read, if we close the connection while
+			// it's trying to upload something that's large.
+			if r.Body != nil {
+				io.Copy(ioutil.Discard, r.Body)
```

Comment:
I could've sworn it was, but I'm having a hard time reproducing it now that we're not pointing the proper client at it in the tests.  I'll drop it for now.